src/table/table.scss

```diff
@@ -85,7 +85,7 @@
       }
     }
 
-    &:hover > .iui-resizer {
+    &:hover:where(:not(.iui-reorder-column-left, .iui-reorder-column-right)) > .iui-resizer {
       opacity: 1;
```

Comment:
Changed to having a separate bar element like resizer is done. This way I don't encounter this resizer and reorder indicator collapse issue also fixes content shifting when hovering over column.
P.S. You can now drag first table's columns because I couldn't work without actual working example 😀 Of course it won't change the order but at least we can test drag interactions.